main/src/ca/uwaterloo/flix/language/phase/typer/TypeReduction.scala

```diff
@@ -293,13 +293,46 @@ object TypeReduction {
       case (Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Array, _), elmType1, _), rcVar1, _),
       Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Array, _), elmType2, _), rcVar2, _)) =>
         isSubtype(elmType1, elmType2)
+      // Arrow to Java function interface
+      case (Type.Apply(Type.Apply(Type.Apply(Type.Cst(TypeConstructor.Arrow(n), _), eff, _), var_arg, _), var_ret, _), Type.Cst(TypeConstructor.Native(clazz), _)) =>
+        val targsVal = tpe1.typeArguments.map(purifyType)
+        targsVal match { // TODO INTEROP: generics support
+          // Pattern-matching Flix functions of the form: eff -> arg -> ret
+          case eff :: TypeConstructor.Native(_) :: TypeConstructor.Native(_) :: Nil =>
```

Comment:
So, if the goal is just to support non-polymorphic functions, then i'd expect `t -> Unit \ eff` to be `Consumer[t]`. @magnus-madsen does any function in java actually use non-polymorphic function interfaces?

Reply:
Maybe we should start with just IntConsumer and the other non-generic function interfaces?

Reply:
I just want to most basic things to work, .e.g IntConsumer